doc/sphinx/testing.rst

```diff
@@ -0,0 +1,419 @@
+
+.. _Unit testing:
+
+Unit testing
+============
+
+Tests are configured in CMake and run by CTest.
+
+
+.. _Cpp unit tests:
+
+C++ unit tests
+--------------
+
+Testing individual C++ functions and classes.
+
+
+Running tests
+^^^^^^^^^^^^^
+
+To build and run the tests:
+
+.. code-block:: bash
+
+   make -j$(nproc) check_unit_tests
+
+
+Writing tests
+^^^^^^^^^^^^^
+
+Framework: `Boost.Test <https://www.boost.org/doc/libs/release/libs/test/>`_.
+
+Store new unit tests in :file:`/src/core/unit_tests/`. Tests usually follow
+this structure:
+
+.. code-block:: c++
+
+   /// @file /src/core/unit_tests/Particle_test.cpp
+   #include <boost/test/unit_test.hpp>
+   #include "core/utils/serialization/Particle.hpp"
+
+   BOOST_AUTO_TEST_CASE(comparison) {
+     {
+       Particle p, q;
+       p.identity() = 1;
+       q.identity() = 2;
+       BOOST_CHECK(p != q);
+     }
+     {
+       Particle p, q;
+       p.identity() = 2;
+       q.identity() = 2;
+       BOOST_CHECK(p == q);
+     }
+   }
+
+
+Mocked classes should be enclosed in a namespace and stored in
+:file:`/src/core/unit_tests/mock`. Here is an example of a mocked Cell:
+
+.. code-block:: c++
+
+   /// @file /src/core/unit_tests/mock/Cell.hpp
+   namespace Testing {
+     template <typename Particle> class Cell {
+       public:
+       Cell() : n(0) {}
+       std::size_t n;
+       std::vector<Particle> part;
+     };
+   } // namespace Testing
+
+which is then used as:
+
+.. code-block:: c++
+
+   #include "mock/Cell.hpp"
+   using Cell = Testing::Cell<Particle>;
+
+
+CMake
+^^^^^
+
+Configure tests in :file:`/src/core/unit_tests/CMakeLists.txt` using the syntax
+
+.. code-block:: cmake
+
+   unit_test(NAME <mytest> SRC <mytest.c> [<../dependency1.c>, ...] [NUM_PROC <N>])
```

Comment:
This is not up-to-date: the `unit_test` cmake function now supports `DEPENDS`, with which cmake targets can be added as dependencies for the test. The way you describe should not be used.

Reply:
Text updated.